kernel/comps/softirq/src/lib.rs

```diff
@@ -119,25 +127,87 @@ static ENABLED_MASK: AtomicU8 = AtomicU8::new(0);
 
 cpu_local_cell! {
     static PENDING_MASK: u8 = 0;
+    static DISABLE_SOFTIRQ_COUNT: u8 = 0;
+}
+
+fn increase_disable_softirq_count() {
+    DISABLE_SOFTIRQ_COUNT.add_assign(1);
+}
+
+fn decrease_disable_softirq_count() {
+    DISABLE_SOFTIRQ_COUNT.sub_assign(1);
+}
+
+fn disable_softirq_count() -> u8 {
+    DISABLE_SOFTIRQ_COUNT.load()
+}
+
+#[clippy::has_significant_drop]
+#[must_use]
+struct DisableLocalSoftirqGuard(PhantomData<()>);
+
+impl Drop for DisableLocalSoftirqGuard {
+    fn drop(&mut self) {
+        // Once the guard is dropped, we will process pending items within
+        // the current thread's context if softirq is going to be enabled.
+        // This behavior is similar to how Linux handles pending softirqs.
+        if disable_softirq_count() == 1 && !in_interrupt_context() {
+            // Preemption and softirq are not really enabled at the moment,
+            // so we can guarantee that we'll process any pending softirqs for the current CPU.
+            process_all_pending();
+        }
+
+        decrease_disable_softirq_count();
+    }
+}
+
+/// Disables softirq on current processor.
+fn disable_softirq_local() -> DisableLocalSoftirqGuard {
+    increase_disable_softirq_count();
+    DisableLocalSoftirqGuard(PhantomData)
+}
+
+/// Checks whether the softirq is enabled on current processor
+fn is_softirq_enabled() -> bool {
+    DISABLE_SOFTIRQ_COUNT.load() == 0
+}
+
+fn get_and_clear_pending() -> u8 {
+    // The PENDING_MASK should be accessed with local interrupts disabled to avoid race conditions,
+    // as the mask is also accessed in the interrupt handler for raising softirqs.
+    let guard = disable_local();
+    let pending_mask = PENDING_MASK.load();
+    PENDING_MASK.store(0);
+    drop(guard);
+    pending_mask
```

Comment:
`drop(guard)` seems redundant.

Why not:
```suggestion
    let _guard = disable_local();

    let pending_mask = PENDING_MASK.load();
    PENDING_MASK.store(0);

    pending_mask
```